### PR TITLE
add resource components

### PR DESCRIPTION
**Pull request by wilsonjr**
* Add dataset resource components

----

> [!IMPORTANT]
> Add new resource components for datasets and embedding sets in `openapi.ts` and create `src/resource.ts` for module imports and UUID type definition.
> 
>   - **New File**:
>     - `src/resource.ts`: Created to import modules like `apache-arrow`, `BaseAtlasClass`, `AtlasIndex`, `AtlasViewer`, and define `UUID` type.
>   - **Models**:
>     - In `openapi.ts`, added `ResourceDataset`, `Resource`, `EmbeddingSetResource`, `ResourceCreationRequest`, and `ResourceCreationResponse` to `components.schemas`.
>     - `Resource` includes `resource_type`, `dependencies`, and `dataset_id`.
>     - `EmbeddingSetResource` specifies `resource_type` as 'EMBEDDING_SET' and includes `embedding_target`.
>     - `ResourceCreationRequest` and `ResourceCreationResponse` handle resource creation parameters and responses.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=nomic-ai%2Fts-nomic&utm_source=github&utm_medium=referral)<sup> for e14a76757392078c0d9819feb433d07e1a0d1c2d. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by wilsonjr** [comment]
* **#84** <a href="https://app.graphite.dev/github/pr/nomic-ai/ts-nomic/84?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/nomic-ai/ts-nomic/84?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/resource.ts
Comment (bmschmidt) [inline src/resource.ts:1]: Let's not merge this component on yet -- instead, we can define it in atlas-next to work off the raw types until we figure out exactly what we need.

File: src/type-gen/openapi.ts
Comment (bmschmidt) [inline src/type-gen/openapi.ts:1]: This is just by running the npm script. right?

Comment (bmschmidt) [inline src/type-gen/openapi.ts:1]: This is just by running the npm script. right?

Comment (wilsonjr) [reply src/type-gen/openapi.ts:1]: Nop, I didn't run the npm script

**Pull request closed without merging**